utils/src/main/scala/za/co/absa/enceladus/utils/performance/PerformanceMetricTools.scala

```diff
@@ -28,15 +28,62 @@ object PerformanceMetricTools {
 
   private val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
+  /**
+    * Adds general job information to Atum's metadata that will end up in an info file.
+    *
+    * The method should be ran at the beginning of a job so that this general information is available for debugging.
+    *
+    * @param spark         A Spark Session
+    * @param optionPrefix  A prefix for all performance metrics options, e.g. 'std' for Standardization and 'conform' for Conformance
+    * @param inputPath     A path to an input directory of the job
+    * @param outputPath    A path to an output directory of the job
+    * @param loginUserName A login user name who performed the job
+    *
+    **/
+  def addJobInfoToAtumMetadata(optionPrefix: String,
+                                inputPath: String,
+                                outputPath: String,
+                                loginUserName: String,
+                                cmdLineArgs: String)
+                               (implicit spark: SparkSession): Unit = {
+    // Spark job configuration
+    val sc = spark.sparkContext
+
+    // The number of executors minus the driver
+    val numberOfExecutors = sc.getExecutorMemoryStatus.keys.size - 1
+
+    val fsUtils = new FileSystemVersionUtils(spark.sparkContext.hadoopConfiguration)
+    // Directory sizes and size ratio
+    val inputDirSize = fsUtils.getDirectorySize(inputPath)
+
+    addSparkConfig(optionPrefix, "spark.driver.memory", "driver_memory")
+    addSparkConfig(optionPrefix, "spark.driver.cores", "driver_cores")
+    addSparkConfig(optionPrefix, "spark.driver.memoryOverhead", "driver_memory_overhead")
+    addSparkConfig(optionPrefix, "spark.executor.memory", "executor_memory")
+    addSparkConfig(optionPrefix, "spark.executor.cores", "executor_cores")
+    addSparkConfig(optionPrefix, "spark.executor.memoryOverhead", "executor_memory_overhead")
+    addSparkConfig(optionPrefix, "spark.submit.deployMode", "yarn_deploy_mode")
+    addSparkConfig(optionPrefix, "spark.master", "spark_master")
```

Comment:
can we somehow pull the `--conf` data as well?

Reply:
Not quite following. Do you mean values such as Spline URL etc?

Reply:
Yes, those. They can have things in that affect the job, right?

Reply:
There are not many config options available at runtime. I've included these that we already use and know they affect performance.
Specifically, JVM options I'm reluctant to include due to security reasons.